app/helpers/tasks_helper.rb

```diff
@@ -0,0 +1,42 @@
+module TasksHelper
+  TASKS_LIMIT = 5
+
+  def assigned_cards(project_id = nil)
+    @assigned_cards ||= begin
+      # Using Arel.sql to sort the records by due_date with null due_date records last
+      cards = current_user.cards.order(Arel.sql('due_date IS NULL, due_date ASC'))
+
+      if project_id
+        cards.select { |card| card.project.id == current_project.id }
```

Comment:
If we're passing project_id we should use it here 
```suggestion
        cards.select { |card| card.project.id == project_id }
```